.github/scripts/transform_dbt_test_results.py

```diff
@@ -676,28 +621,119 @@ def _filter_for_existing_field_indexes(
         )
 
 
+# Help docstring for the command line interface
+CLI_DESCRIPTION = """Generates an Excel workbook of dbt test failures that can be shared with other teams for review and correction,
+along with metadata parquet files that can be uploaded to S3 for long-term result tracking.
+
+This script assumes that it is being run in sequence after a call to `dbt test --store-failures`, since it depends on two files created
+by that operation (target/run_results.json and target/manifest.json). It also requires Python dependencies be installed from
+requirements.transform_dbt_test_results.txt.
+
+Expects one required environment variable to be set:
+
+ 1. USER: The username of the user who ran the script. This is automatically set during login on Unix systems, but should be set manually elsewhere.
+
+Expects four optional environment variables:
+
+ 1. AWS_ATHENA_S3_STAGING_DIR: Location in S3 where Athena query results should be written (defaults to s3://ccao-athena-results-us-east-1)
+ 2. GIT_SHA: The SHA of the latest git commit (defaults to the output of `git rev-parse`)
+ 3. GIT_REF: The name of the ref for the latest git commit (defaults to the output of `git rev-parse --abbrev-ref`)
+ 4. GIT_AUTHOR: The author of the latest git commit (defaults to the output of `git log`)
+
+Outputs three files to the directory specified by the `--output-dir` flag:
+
+  1. `qc_test_failures_<date>.xlsx`: Excel workbook to share with other teams
+  2. `metadata/test_run/run_year=YYYY/*.parquet`: Metadata about this run, partitioned by year of run and prepped for upload to S3
+  3. `metadata/test_run_result/run_year=YYYY/*.parquet`: Metadata about test results (pass, fail, number of failing rows, etc.) in this run,
+     partitioned by year of run and prepped for upload to S3
+
+Each sheet in the output workbook represents a category of test, e.g. "valid_range" or "not_null"; each row in a sheet represents a row in
+a database that failed a test, with enough metadata that a reader can figure out what conditions caused the test to fail and investigate the root cause."""  # noqa: E501
```

Comment:
The lines here are slightly longer than the typical flake8 limit of 80 characters for better terminal UX.